zipline/algorithm.py

```diff
@@ -114,6 +115,38 @@
 DEFAULT_CAPITAL_BASE = float("1.0e5")
 
 
+log = logbook.Logger("ZiplineLog")
+
+
+def check_asset_lifetime(f):
+    @wraps(f)
+    def wrapper(*args, **kwargs):
+        trading_algorithm = args[0]
```

Comment:
See note below regarding sid/asset naming inconsistency in our other order methods.